src/rdiff_backup/SetConnections.py

```diff
@@ -169,11 +169,19 @@ def init_connection(remote_cmd):
     try:
         # we need buffered read on SSH communications, hence using
         # default value for bufsize parameter
-        process = subprocess.Popen(
-            remote_cmd,
-            shell=True,
-            stdin=subprocess.PIPE,
-            stdout=subprocess.PIPE)
+        if os.name == 'nt':
```

Comment:
Just to avoid repetition of code:
```
remote_cmd = os.fsdecode(remote_cmd) if os.name == 'nt' else remote_cmd
process = subprocess.Popen(
...
```
Or something similar.

Reply:
the reason for the duplication is that the string is used in further commands down the line and hence I didn't want to simply overwrite its byte version with its string version. I could have worked with a temporary variable but that wouldn't have made the code nicer IMHO.